crypto/sm4/asm/sm4-riscv64-zvksed.pl

```diff
@@ -59,6 +59,524 @@
 .text
 ___
 
+my ($V0, $V1, $V2, $V3, $V4, $V5, $V6, $V7,
+    $V8, $V9, $V10, $V11, $V12, $V13, $V14, $V15,
+    $V16, $V17, $V18, $V19, $V20, $V21, $V22, $V23,
+    $V24, $V25, $V26, $V27, $V28, $V29, $V30, $V31,
+) = map("v$_",(0..31));
+
+# Load 32 round keys to v1-v8 registers.
+sub enc_load_key {
+    my $keys = shift;
+
+    my $code=<<___;
+    @{[vsetivli "zero", 4, "e32", "m1", "ta", "ma"]} 
+    @{[vle32_v $V16, $keys]} # rk[0:3]  
+    addi $keys, $keys, 16
+    @{[vle32_v $V17, $keys]} # rk[4:7]
+    addi $keys, $keys, 16
+    @{[vle32_v $V18, $keys]} # rk[8:11]
+    addi $keys, $keys, 16
+    @{[vle32_v $V19, $keys]} # rk[12:15]
+    addi $keys, $keys, 16
+    @{[vle32_v $V20, $keys]} # rk[16:19]
+    addi $keys, $keys, 16
+    @{[vle32_v $V21, $keys]} # rk[20:23]
+    addi $keys, $keys, 16
+    @{[vle32_v $V22, $keys]} # rk[24:27]
+    addi $keys, $keys, 16
+    @{[vle32_v $V23, $keys]} # rk[28:31]    
+___
+
+    return $code;
+}
+
+sub dec_load_key {
+    my $keys = shift;
+
+    my $code=<<___;
+    @{[vle32_v $V23, $keys]} # rk[31:28]
+    addi $keys, $keys, 16
+    @{[vle32_v $V22, $keys]} # rk[27:24]
+    addi $keys, $keys, 16
+    @{[vle32_v $V21, $keys]} # rk[23:20]
+    addi $keys, $keys, 16
+    @{[vle32_v $V20, $keys]} # rk[19:16]
+    addi $keys, $keys, 16
+    @{[vle32_v $V19, $keys]} # rk[15:12]
+    addi $keys, $keys, 16
+    @{[vle32_v $V18, $keys]} # rk[11:8]
+    addi $keys, $keys, 16
+    @{[vle32_v $V17, $keys]} # rk[7:4]
+    addi $keys, $keys, 16
+    @{[vle32_v $V16, $keys]} # rk[3:0]   
+___
+
+    return $code;
+}
+
+# sm4 encryption with round keys v1-v8
+sub enc_blk {
```

Comment:
Similarly, simplify `rv64i_zvksed_sm4_encrypt` using `enc_blk`.

The comment here should be round keys v16-v23, not v1-v8. If you use a consistent register naming inside this file, that should be round keys vk0-vk7.